### PR TITLE
base & test - status

### PR DESCRIPTION
**Pull request by ttodua**

### CODE REVIEW

**Comment by carlosmiei** [comment]
LGTM ✅

**Comment by carlosmiei** [comment]
@ttodua this change breaks the build, can you adjust the test?

File: ts/src/base/Exchange.ts [hunk 1]
```diff
@@ -3029,10 +3029,19 @@ export default class Exchange {
...
-                'info': time,
-            });
+            try {
+                this.status = this.extend (this.status, {
+                    'status': 'ok',
+                    'updated': time,
+                    'info': time,
+                });
+            } catch (ex) {
+                this.status = this.extend (this.status, {
```
Comment (ttodua) [inline ts/src/base/Exchange.ts:3039]: It was a bit questionable, but to see from the following point:
- in userland code user checks for status, and the emulated `fetchStatus` , specifically `fetchTime` does not succeed - it should mean the failure of status. let's assume user has wrapped try/catch on top of it, and it means, the 'status' property is being left still with 'ok' status, even though this last call would be failing. So, when `fetchTime` fails, we set status: fail.

**Comment by ttodua** [comment]
Now this passes build.

**Comment by ttodua** [comment]
> @ttodua this change breaks the build, can you adjust the test?

build passes here now

**Pull request closed without merging**